spyder/plugins/ipythonconsole/widgets/shell.py

```diff
@@ -454,18 +458,68 @@ def change_mpl_backend(self, command):
                 self.silent_execute(command)
 
     # ---- Spyder-kernels methods -------------------------------------------
+    def get_editor(self, filename=None):
+        """Get editor for filename and set it as the current editor."""
+        editorstack = self.get_editorstack()
+        if editorstack is None:
+            return None
+
+        if not filename:
+            return editorstack.get_current_editor()
+
+        index = editorstack.has_filename(filename)
+        if index is None:
+            return None
+
+        editor = editorstack.data[index].editor
+        editorstack.set_stack_index(index)
+        return editor
+
     def get_editorstack(self):
         """Get the current editorstack."""
         plugin = self.ipyclient.plugin
         if plugin.main.editor is not None:
             editor = plugin.main.editor
             return editor.get_current_editorstack()
+        raise RuntimeError('No editorstack found.')
```

Comment:
Why the need to raise this error here?

Reply:
This error and the three errors below are caused when the kernel is calling functions, so possiblt it would want to do something else than show an error.

Reply:
Like: save this file, and if it is not possible, I will execute it anyway.

Reply:
So are these errors handled in the kernel?

Reply:
They are displayed in the console